ktor-client/ktor-client-plugins/ktor-client-content-negotiation/common/src/io/ktor/client/plugins/contentnegotiation/ContentNegotiation.kt

```diff
@@ -77,11 +92,84 @@ public class ContentNegotiation internal constructor(
             registrations.add(registration)
         }
 
+        /**
+         * Adds a type to the list of types that should be ignored by [ContentNegotiation].
+         *
+         * The list contains the [HttpStatusCode], [ByteArray], [String] and streaming types by default.
+         */
+        public inline fun <reified T> ignoreType() {
```

Comment:
does it make sense to have API with `KClass`? (this will also allow us to make `ignoredTypes` not `PublishedApi`

Reply:
Could you tell me why we need KClass API and why PublishedApi is not ok?

Reply:
1. KClass (or even KType) Api is sometimes needed, when you want to dynamically configure ignored types. Even internally `DefaultCommonIgnoredTypes` and `DefaultIgnoredTypes` are sets of types, which you want to ignore in not reified manner. So I would expect, that users will also want at some day do something like this. There were already several times issue raised in youtrack in routing, that ktor only has reified functions.
2. `PublishedApi` is tracked in binary compatibility. So if later `ignoreType` underlying representation will change and f.e. `ignoredTypes` will be not Set, or we will add additional parameter to `ignoreType` which we will need to store, we should keep `ignoredTypes` there, working, as it's can be referenced from `public` api because of inline functions, that were already in user code and compiled. So it makes internal representation effectively public, which is not good.

Reply:
Agree! Let's add KClass API